shared/src/error/mod.rs

```diff
@@ -1,101 +1,116 @@
-// Copyright 2019-2022 ChainSafe Systems
-// SPDX-License-Identifier: Apache-2.0, MIT
-
 use std::fmt::Formatter;
 
 use num_derive::FromPrimitive;
+use serde::{Deserialize, Serialize};
 use thiserror::Error;
 
-use crate::encoding::repr::*;
-
-/// ExitCode defines the exit code from the VM execution.
-#[repr(u32)]
-#[derive(
-    PartialEq, Eq, Debug, Clone, Copy, FromPrimitive, Serialize_repr, Deserialize_repr, Error,
-)]
-pub enum ExitCode {
-    Ok = 0,
-
-    /// Indicates failure to find an actor in the state tree.
-    SysErrSenderInvalid = 1,
+/// ExitCode defines the exit code from the VM invocation.
+#[derive(PartialEq, Eq, Debug, Clone, Copy, Serialize, Deserialize)]
+#[serde(transparent)]
+pub struct ExitCode {
```

Comment:
We should also mark this as `#[repr(transparent)]`. That will ensure that the rust compiler represents it as a `u32` (in memory) and will make it safe to use in syscalls (as arguments & return values).